Documentation update

Jars-of-jam-Scheduler opened the pull request:
| Q | R |
| --- | --- |
| Correction de bugs ? | non |
| Nouvelle Fonctionnalité ? | non |
| Tickets (_issues_) concernés | - |

Cette mise à jour de la documentation comprend :
1. L'ajout du fichier "presentation-install.rst", qui présente de manière générale les différents outils à installer pour pouvoir contribuer à ZdS ;
2. Ajout des paquets Fedora ;
3. Ajout d'explications sur certains outils ;
4. Ajout de commandes pour pouvoir les utiliser.

Attention : certaines commandes bash ne sont pas mises en forme, et je ne comprends pas pourquoi...

Jars-of-jam-Scheduler committed:
Create presentation-install.rst

This document describes the main utils you need to work with ZdS.

Jars-of-jam-Scheduler committed:
Update backend-linux-install.rst

Adding :
- Fedora's packages
- Commands
- Commands' explanations

Jars-of-jam-Scheduler committed:
Update frontend-install.rst

Addings :
1. Gulp

Updates :
1. npm&node.js installation

Jars-of-jam-Scheduler committed:
Update presentation-install.rst

Jars-of-jam-Scheduler committed:
Update presentation-install.rst

Minor update.

Jars-of-jam-Scheduler committed:
Update presentation-install.rst

Jars-of-jam-Scheduler committed:
Update presentation-install.rst

Jars-of-jam-Scheduler committed:
Update backend-linux-install.rst

Jars-of-jam-Scheduler committed:
Update backend-linux-install.rst

pierre-24 commented:
Bienvenue dans le monde de la contribution à ZdS :)

Bon, j'ai tout de même quelques remarques à faire, mais ne le prend pas forcément pour toi (c'est la première fois, et la première PR est toujours difficile)
- Tout ces _commit_ devront au final ne faire qu'un
- La documentation n'est pas écrit en markdown, mais en ReST. Ce qui signifie que toute les lignes de codes doivent commencer par 4 espaces, sinon, [comme tu peux le voir ici](https://github.com/Lern-X/zds-site/blob/documentation_update/doc/source/install/backend-linux-install.rst), ça ne marche pas.
- N'hésite pas à faire des liens (au sens web du terme): quand tu met "voir la partie machin", met un lien vers le fichier correspondant :)
- Pour les titres, il faut autant de `=` ou de `-` que de caractères dans le titre (ReST, une fois encore)
- N'hésite pas à installer chez toi l'environement de ZdS afin de pouvoir générer la documentation. Malgré le fait que certaines parties lancent actuelement des erreurs, tu ne devrais pas être affecté !

Bonne chance !

Eskimon commented on doc/source/install/backend-linux-install.rst:
```diff
@@ -26,50 +26,188 @@ Assurez vous que les dépendances suivantes soient résolues :
+
+.. sourcecode:: bash
+
+    sudo pip install virtualenv
```
Pourquoi ca devient `sudo` ?

SpaceFox replied:
Je pense même qu'il ne faut _pas_ que ce soit fait avec `sudo`.

pierre-24 replied:
Bien vu :)

pierre-24 replied:
... Ou pas, en fait. On peut se passer de `sudo` quand on est dans virtualenv. Or, ici, on y est pas, et pour moi, c'est la seule manière d'installer virtualenv (je viens de vérifier)

Jars-of-jam-Scheduler replied:
pip install <paquet> ne semble pas être utilisable en tant que non-su en tout cas.

yapper-git replied:
Euh si c'est au moins possible avec `pip install --user` sauf erreur

SpaceFox commented on doc/source/install/backend-linux-install.rst:
```diff
@@ -26,50 +26,188 @@ Assurez vous que les dépendances suivantes soient résolues :
+- libjpeg
+- libjpeg-devel
+- freetype
+- freetype-devel
```
Si on commence à multiplier les listes, un tableau est sans doute plus pertinent.

Situphen replied:
On peut-être même un fichier à part ?

Jars-of-jam-Scheduler replied:
Situphen, je pense qu'un tableau serait suffisant. :) Je vais voir comment faire.

Au fait, est-ce que c'est nécessairement moi qui dois faire une modif' quand il en faut une ou vous en avez la possibilité par défaut ?

pierre-24 replied:
C'est ta branche sur ton dépot, donc personne d'autre que toi n'as la possibilité de faire la modif' (par défaut). D'où le nombre important de commentaire :)

Jars-of-jam-Scheduler replied:
Ca marche, je mettrai à jour tout cela demain :-)

Jars-of-jam-Scheduler replied:
Quelqu'un peut me dire comment créer un tableau avec ReST s'il vous plaît ?
Au passage, je n'ai trouvé aucune info sur Google au sujet de cette techno ; existe-t-il un tuto ?

pierre-24 replied:
C'est ma faute, je n'utilise pas le bon nom : 
http://docutils.sourceforge.net/rst.html (ou de manière plus rapide: 
http://docutils.sourceforge.net/docs/user/rst/quickref.html#tables)

On 05/14/2015 09:20 AM, Lern-X wrote:

> In doc/source/install/backend-linux-install.rst 
> https://github.com/zestedesavoir/zds-site/pull/2663#discussion_r30319403:
> 
> > -Ou, en une ligne,
> > +Si vous utilisez la distribution Fedora, ces noms de paquets changent. Considérez ainsi la liste suivante :
> > +
> > +- git
> > +- python
> > +- python-devel
> > +- python-setuptools
> > +- geoip
> > +- geoip-devel
> > +- libxml-devel
> > +- libxslt-devel
> > +- zlib-devel
> > +- libjpeg
> > +- libjpeg-devel
> > +- freetype
> > +- freetype-devel
> 
> Quelqu'un peut me dire comment créer un tableau avec ReST s'il vous 
> plaît ?
> Au passage, je n'ai trouvé aucune info sur Google au sujet de cette 
> techno ; existe-t-il un tuto ?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/zestedesavoir/zds-site/pull/2663/files#r30319403.

Jars-of-jam-Scheduler replied:
Il n'y a vraiment pas un moyen plus rapide de générer un tableau via rst ? Pourquoi vous utilisez cette techno ? J'ai l'impression que tout est plus long à faire avec cet outil : pour créer des titres, il faut avoir le même nombre de = ou de - que de caractères composant le titre, pour générer un tableau, il faut utiliser des -, des | et des + en pagaille pour le recréer visuellement...

Quel est l'intérêt de Rst ? :)

pierre-24 replied:
On utilise cette techno parce que c'est celle qui fonctionne avec Sphinx (le générateur de doc de Zeste de Savoir). Certes, elle est un peu plus exigante, mais elle permet de faire plus de truc (vois ça plus comme LaTeX que comme le _markdown_).

Jars-of-jam-Scheduler replied:
Ok ; sais-tu s'il existe un générateur de tableaux rst ? J'ai trouvé des générateurs Markdown, et Latex, mais pas rst...

pierre-24 replied:
La syntaxe de tableau _grid-table_ de Markdown est celle de RST aussi, non ?

Jars-of-jam-Scheduler replied:
Pas tout-à-fait justement :-/

Situphen commented on doc/source/install/frontend-install.rst:
```diff
@@ -6,6 +6,16 @@ Vous voulez nous aider au développement du frontend ? Installez Node.js et npm
+- node.js (couche logicielle installable sur un serveur Web pour utiliser le langage JavaScript côté-serveur),
+- npm (gestionnaire de modules de node.js – car oui, node.js est modulaire : on peut étoffer ses fonctionnalités en installant divers modules).
+
+Normalement et en toute rigueur, ces dépendances sont à installer dans un environnement virtuel JavaScript. Cependant, son existence n'a pas encore été décidée par l'équipe de développement de Zeste de Savoir. Rappelons que le seul environnement virtuel existant est dédié au serveur Python et à ses dépendances qui sont des paquets exclusivement Python. Ainsi, il ne faut pas y installer de paquet  ...
```
Ta définition de Node.JS est correcte mais vu que dans ce projet on n'utilise pas Node.JS en temps de serveur, je dirais plutôt que c'est un interpréteur de Javascript (définition plus globale).

Pourquoi voudrais-tu que l'on fasse installer un environnement virtuel pour Node.JS et npm aux devs ? Les devs front peuvent en installer un s'ils veulent (`nvm`) mais pour les backs c'est pas très utile ! De plus, pas besoin de ça pour les autres dépendances installées avec npm car elles le sont localement (dans `node_modules/`).

Jars-of-jam-Scheduler replied:
A quoi sert Node.js pour le serveur précisément ? Sinon, la définition que je donne du module me semble plus générale que la tienne, et également plus exacte (Node.js possède un interpréteur JS).

Concernant l'EV : c'est juste que c'est plus propre (on le fait bien pour tout ce qui est Python). Qu'on soit dév back ou front end, ça ne change pas grand-chose si on aime la propreté haha :-)

Situphen replied:
Node.JS ne nous sert qu'à générer les fichiers JS, CSS et les images. Il n'y a pas de lien avec le serveur python. Concernant la définition, on peut reprendre wikipédia : "Node.js est une plateforme logicielle libre et événementielle en JavaScript" https://fr.wikipedia.org/wiki/Nodejs

Je trouve pas ça beaucoup plus propre car de base npm installe les dépendances en local. @Sandhose, tu es penses quoi ?

Jars-of-jam-Scheduler replied:
Comment ça, il génère des fichiers JS, CSS et des images ? Je vois mal comment un framework/une librairie JS à la base conçue pour exécuter du JS (surtout sur serveur) peut faire cela, et en quoi c'est utile. Peut-être que tu confonds avec Gulp (qui ne génère rien du tout en lui-même) ?

Après je me trompe peut-être hein, mais je ne comprends décidémment pas à quoi sert Node.js si cet outil n'est pas utilisé par le serveur ou sur le serveur.

Je veux bien reprendre la définition de Wikipédia, mais elle est un peu insipide, tu ne trouves pas ?

Concernant les npm : oui, c'est vrai, de toute façon on peut retirer ce passage sur l'EV front-end. :-)

Situphen replied:
Node.JS nous sert à interpréter/exécuter/faire fonctionner des outils (comme Gulp) écrits en JS. Certes, il peut servir de serveur mais ici ce n'est pas notre cas !

Jars-of-jam-Scheduler replied:
Ok, pas de soucis :-)
Du coup je préfère supprimer la partie où je parle d'un environnement virtuel pour les outils front-end (même si je continue de penser que ça aurait pu être "utile" pour les gens maniaques, disons).

Concernant la définition de Node.js, je propose de faire un mixte entre celle donnée par Wikipédia et celle que tu viens de donner à l'instant.

Situphen commented on doc/source/install/frontend-install.rst:
```diff
@@ -6,6 +6,16 @@ Vous voulez nous aider au développement du frontend ? Installez Node.js et npm
+
+Normalement et en toute rigueur, ces dépendances sont à installer dans un environnement virtuel JavaScript. Cependant, son existence n'a pas encore été décidée par l'équipe de développement de Zeste de Savoir. Rappelons que le seul environnement virtuel existant est dédié au serveur Python et à ses dépendances qui sont des paquets exclusivement Python. Ainsi, il ne faut pas y installer de paquet  ...
+
+Il est donc nécessaire d'installer node.js et npm directement sur votre distribution Fedora. Si vous refusez catégoriquement de la polluer et que vous souhaitez ardement contribuer au projet Zeste de Savoir, nous vous invitons à nous rejoindre ou à nous contacter afin de mettre en place avec nos développeurs un environnement virtuel JavaScript.
```
Pour moi cette partie est inutile mais si tu veux dire qu'il existe la possibilité d'avoir plusieurs versions de Node.JS et npm sur la même distribution, tu peux.

Pourquoi parler de Fedora à cette endroit ?

Jars-of-jam-Scheduler replied:
La partie qui suit les courtes descriptions des deux modules peut en effet être enlevée (je l'avais mise parce que je m'étais souvenu m'être demandé pourquoi un EV JS n'avait pas été choisi).

Situphen commented on doc/source/install/frontend-install.rst:
```diff
@@ -74,8 +84,7 @@ Il vous faut tout simplement faire :
 
-    sudo curl -sL https://rpm.nodesource.com/setup | bash -
-    sudo yum install -y nodejs
+    sudo dnf install -y nodejs
```
`dnf` est le gestionnaire de dépendances de base ?

Jars-of-jam-Scheduler replied:
Yep, depuis la dernière mise-à-jour, Fedora 22 (qui sort à la fin du mois).

Situphen replied:
Ce serait bien de laisser les anciennes commandes pour ceux qui n'ont pas la dernière version de Fedora. "Pour Fedora < 22..." Tu vois ?

Jars-of-jam-Scheduler replied:
Pas de problème.
Je laisserai toutefois une note à l'attention de l'internaute pour l'inciter à passer à la V22 et donc à utiliser dnf.

Situphen commented on doc/source/install/frontend-install.rst:
```diff
@@ -173,6 +182,14 @@ Si vous voulez utiliser directement la commande ``gulp [tâche]`` au lieu de ``n
+    
+  Rendez-vous dans votre fork local avec la commande « cd ». Il s'agit désormais de générer les fichiers-sources du front avec Gulp, la boîte à outils !
+
+Commande de génération du front-end :
```
Je ne pense pas que ce soit le bon endroit pour mettre ça. D'ailleurs, je suis en train de modifier cette partie avec #2641 !

Jars-of-jam-Scheduler replied:
Bein Gulp doit logiquement être : 
1. Dans la partie front-end
1.1. Après l'install de node.js et npm
2. Après le fork (back-end||front-end)

Qu'en penses-tu ? Où aurais-tu souhaité mettre ça ?

Situphen replied:
Je l'aurais mis là où on parle des différentes tâches de Gulp, c'est-à-dire un peu plus haut. Comme je l'ai dit, j'ai réécrit cette partie dans #2641 (voir https://github.com/Situphen/zds-site/blob/improve-front-end-doc/doc/source/install/frontend-install.rst#utilisation-des-outils) ! ;)

Jars-of-jam-Scheduler replied:
J'ai lu ton document. Dans ce cas, il faut, dans la partie back-end, inciter l'internaute à forker avant de lui demander d'aller faire un tour côté front-end :-)

Situphen replied:
Vu que tu le fais avec cette _pull request_, je n'ai pas besoin de le faire ! #ThePowerOfGit

Situphen commented:
Désolé pour la froideur de mes commentaires. Comme l'a dit @pierre-24, il ne faut pas le prendre personnellement ! :)

En tous cas, bravo pour cette PR et bienvenue dans l'équipe de dev' ! ;)

pierre-24 commented on doc/source/install/presentation-install.rst:
```diff
@@ -0,0 +1,32 @@
+L'environnement de travail est constitué par… :
+
+-Les sources de Zeste de Savoir,
+
```
Ne met pas d'espace entre les lignes d'une liste, ReST n'as pas l'air d'apprécier ^^

pierre-24 commented on doc/source/install/presentation-install.rst:
```diff
@@ -0,0 +1,32 @@
+
+-Un environnement virtuel* qui contient le serveur précédemment cité.
+
+*: notion expliquée ci-dessous.
```
Met un `\` avant ton *, sinon ReST le considère bizarrement

Jars-of-jam-Scheduler replied:
Pas de soucis, j'apporterai ces modifs' demain :)

SpaceFox commented:
Bon, j'ai pas le temps de relire tout ça en détail, mais globalement, il faut se rappeler qu'ici tu mets à jour une **documentation**. Une documentation, c'est un texte qui doit donner le maximum d'information le plus clairement possible.

Ça implique en particulier que seuls 3 éléments sont autorisés dans une doc :
1. Une information qui s'applique directement (_"Pour installer ZdS, vous devez faire ça"_) (ça peut être un conseil fort, par exemple : _"vous devriez installer virtualenv"_)
2. L'explication pour comprendre ce qu'on doit faire (_"on fait ceci parce que..."_)
3. Des ressources externes pour les détails qui ne nous concernent pas directement (_"Pour aller plus loin, allez voir..."_)

Tout le reste peut partir à la poubelle. Notamment toutes les considérations du style "normalement...", les enjeux évidents, les duplicatas de docs (exemple : comment générer une paire de clés SSH, la doc Github à ce sujet est très bien), ...

Jars-of-jam-Scheduler commented:
Ok, SpaceFox :-)
Je prends en compte les différentes remarques qui ont été faites et je vais remanier tout ça aujourd'hui.

Jars-of-jam-Scheduler committed:
Update frontend-install.rst

Suppressions :
1. Les deux paragraphes parlant d'un environnement virtuel JavaScript ;

Modifications :
1. La bève description de Node.js a été remaniée ;
2. Commande d'installation sous Fedora : utilisation de yum par défaut (mais mention de dnf pour v>=22)

Concernant Gulp :
Situphen est en train de remanier cette partie, je préfère donc ne pas y toucher. Il faudrait de préférence que le clonage du dépôt se fasse dans la page "backend", avant la partie invitant l'internaute à installer le front-end (puisque la partie front-end nécessite le fork à un moment donné : Gulp).

Jars-of-jam-Scheduler committed:
Update frontend-install.rst

Ajout de deux espaces ("Nettoyage des outils" > "Désinstaller les dépendances" > Commande "npm uninstall".

Jars-of-jam-Scheduler commented:
@Situphen  : recherche Google avec le mot "être" dedans : https://www.google.fr/search?q=laissez-vous+guider&ie=utf-8&oe=utf-8&gws_rd=cr&ei=ja5UVZuQG4X8UselgXg#q=laissez-vous+%C3%AAtre+guider et en résultats, que des "Laissez-vous guider", donc nope c'est bien "Laissez-vous guider" :-)
- Pourquoi ne puis-je pas directement répondre à ton commentaire ?

Jars-of-jam-Scheduler committed:
Update backend-linux-install.rst

1. Quatre espaces ont été placés avant les lignes de code ;
2. Quelques suppressions (clés SSH principalement).

Jars-of-jam-Scheduler committed:
Update presentation-install.rst

Suppressions des retours-chariot de la liste et échappement de l'astérisque.

Jars-of-jam-Scheduler committed:
Update presentation-install.rst

Minor changes

firm1 commented on doc/source/install/backend-linux-install.rst:
```diff
@@ -26,51 +26,170 @@ Assurez vous que les dépendances suivantes soient résolues :
+- python
+- python-devel
+- python-setuptools
+- geoip
```
Normalement, on a plus besoin d'installer `geoip` et `geoip-devel`.

firm1 commented on doc/source/install/backend-linux-install.rst:
```diff
@@ -26,51 +26,170 @@ Assurez vous que les dépendances suivantes soient résolues :
+    sudo dnf install git python python-devel python-setuptools
+    sudo easy_install pip
+    sudo pip install tox
+    sudo dnf install geoip geoip-devel libxml-devel libxslt-devel zlib-devel libjpeg libjpeg-devel freetype freetype-devel
```
Idem ici, plus besoin d'installer `geoip` et `geoip-devel`

Jars-of-jam-Scheduler replied:
Ok, je vais modifier ça, merci :-)

pierre-24 commented:
À la fin de tout ce travail (qu'on te remercie d'avance de faire, d'ailleurs, une fois encore fait pas attention au ton de nos commentaires), n'hésite pas à te renseigner sur la manière de mettre tout tes _commits_ en un (ou à passer sur IRC demander ;) )

Situphen commented:
On a justement un tutoriel sur ZdS avec un extrait sur comment [fusionner des _commits_](https://zestedesavoir.com/tutoriels/649/refaire-lhistoire-avec-git/#3-fusionner-des-commits) !

Jars-of-jam-Scheduler committed:
Update backend-linux-install.rst

Jars-of-jam-Scheduler committed:
Update backend-linux-install.rst

Jars-of-jam-Scheduler committed:
Update backend-linux-install.rst

Jars-of-jam-Scheduler committed:
Update backend-linux-install.rst

Eskimon commented:
On est bon ici ou pas ? Si oui il faudrait squash/fixup tout les commits pour que ca soit plus propre, un petit rebase et en avant guingamp !

Eskimon commented:
ping @Lern-X :)

Jars-of-jam-Scheduler commented:
Salut tout le monde,

Désolé pour ce petit moment d'inactivité, j'ai eu quelques soucis avec ma Fedora que j'ai dû réinstaller et j'ai été très occupé par mes études. Cela dit, ça devrait aller mieux maintenant.

Je vais rassembler les commits du coup, après avoir re-forké le dépôt.

GerardPaligot commented:
@Lern-X Est-ce que tu as des nouvelles ici ?

GerardPaligot commented:
@SpaceFox et @Eskimon, @Lern-X ne semble plus présent. On fait quoi de cette PR ?

SpaceFox commented:
S'il suffit de rebase / squash des commits et que le fond est bon, on peut la reprendre non ?

GerardPaligot commented:
Difficile à dire pour moi, je suis sur OS X et il y a beaucoup de changement dans la section Linux. Si ses changements sont ok, je peux m'occuper du rebase/squash.

Jars-of-jam-Scheduler commented:
Salut tout le monde,

Désolé de ne pas avoir été très actif ces derniers mois, le temps m'a manqué, ma Fedora a bogué suite à l'installation de la nouvelle version... et je n'avais pas réussi à ré-installer l'environnement de travail de ZdS.

Je suis passé sous Kubuntu, et maintenant tout semble fonctionner correctement (l'EDT, ma distro...). Du coup j'ai re-forké le dépôt, ai récupéré mon travail et je suis en train de vérifier les commandes que j'avais écrites dans la doc.

Vous devriez peut-être supprimer cette PR, puisque du coup j'en referai une autre si vous n'y voyez pas d'inconvénient.

Eskimon commented:
Ok je ferme à ta demande. Elle pourra toujours être rouverte au besoin ou cherry-pick pour reprendre le travail :)
